classes/tools/mattermost_tools.php

```diff
@@ -936,10 +936,10 @@ public static function synchronize_groups($courseid) {
                         // At end of the channel name.
                         $mattermostchannelid =
                             $mattermostapimanager->create_mattermost_channel($channelname . '_' . time());
+                    } else {
```

Comment:
This same mistake is in observers also. Check if there is any other place where this mistake is present.

Reply:
Fixed, these are the only two places where we had this.